### ISSUE TITLE
Add support for pinned packages

### ISSUE THREAD

**Issue by rejeep** [cask/cask#252]
package.el in Emacs 24.4 includes a new feature to pin packages (see `package-pinned-packages`). Provide a way for Cask to pin packages.

For example:

``` lisp
(depends-on "foo")
(depends-on "bar" :archive "melpa-stable")
```

**Comment by nlamirault** [comment]
Does Cask support pinned packages ?

**Comment by rejeep** [comment]
@nlamirault Nope, not yet.

**Comment by konukhov** [comment]
Any updates on this? Would be pretty neat to have pinned packages support in Cask.

**Comment by rejeep** [comment]
Nope. I'm sorry to say, but I don't have a lot of time for open source these days, so unless someone else contributes this feature, it's likely it will end up in Cask.

**Comment by konukhov** [comment]
@rejeep Thanks for the answer! I hope some day someone (even me, but I don't have time either) will implement this.

**Comment by rmuslimov** [comment]
+1 for this feature

**Comment by hanjaelee** [comment]
+1 I need this feature.

**Comment by dgellow** [comment]
Should be solved by #355 but only if you're using emacs v24.4+.

**Comment by dickmao** [comment]
Pinning to an archive is not as relevant as it was in 2014-2016 when melpa-stable was a viable option.  It really is no longer, especially for the kind of advanced user that would employ cask.
